### PR TITLE
Deploy to Google Cloud

### PR DESCRIPTION
**Pull request by codetheweb**
What's missing:

- a bucket for user files
- OAuth client set up
- email service

### CODE REVIEW

File: src/backend/scripts/run-with-gcp-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { execa } from "execa";
+import * as gcpMetadata from "gcp-metadata";
+import axios from "axios";
+
+/**
+ * This script injects Google Cloud metadata (when available) into the environment.
+ * Specifically, it injects the NEXTAUTH_URL and BASE_HTTP_ENDPOINT environment variables.
+ * Run with `npm run with-gcp-metadata <command> <args>`.
+ */
```
Comment (codetheweb) [inline src/backend/scripts/run-with-gcp-metadata.ts:9]: very frustrating: the generated cloud run URL isn't known until after the deploy, and terraform blocks can't reference themselves

I think this is the cleanest solution, didn't see anything simpler online

Comment (amantri) [reply src/backend/scripts/run-with-gcp-metadata.ts:9]: NEXAUTH_URL will be our domain, e.g. [staging.]compassiep.com

File: terraform/gcs/cloud_run.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+      ports {
+        container_port = 3000
+      }
+
+      env {
+        name = "DATABASE_URL"
+        value_source {
+          secret_key_ref {
+            secret  = google_secret_manager_secret.database_url.secret_id
```
Comment (amantri) [inline terraform/gcs/cloud_run.tf:19]: very cool!

File: terraform/gcs/cloud_sql.tf [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+resource "google_sql_database_instance" "postgres_instance" {
+  name             = "postgres-instance"
+  region           = "us-west1"
+  database_version = "POSTGRES_15"
+
+  settings {
+    tier = "db-f1-micro"
+  }
+
+  # todo: set `deletion_protection` depending on the environment
```
Comment (amantri) [inline terraform/gcs/cloud_sql.tf:11]: 🐧
```suggestion
  # TODO: set `deletion_protection` depending on the environment
```
Makes it more prominent (and also consistent style)

**Review by amantri — APPROVED** [review]

File: src/backend/scripts/run-with-gcp-metadata.ts (hunk 1 above)
Comment (codetheweb) [reply src/backend/scripts/run-with-gcp-metadata.ts:9]: yes, however ideally we can deploy without a domain/with the generated one with necessary
it'd be nice to be able to deploy to a new project or a personal account for testing

File: terraform/gcs/main.tf [hunk 4]
```diff
@@ -0,0 +1,17 @@
+terraform {
```
[the remark this answers is not in the document]
Comment (codetheweb) [reply terraform/gcs/main.tf:1]: c5a060f

**Pull request merged**